File: src/main/java/org/vaadin/example/Binder/FormButtonsBar.java
package org.vaadin.example.Binder;



import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.templatemodel.TemplateModel;

@Tag("form-buttons-bar")
@JsModule("./src/js/form-buttons-bar.js")
@Route("formButtonsBar")
public class FormButtonsBar extends PolymerTemplate<TemplateModel> {

    /**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id("saveButton")
    private Button saveButton;

    @Id("cancelButton")
    private Button cancelButton;

    @Id("deleteButton")
    private Button deleteButton;

    public FormButtonsBar() {
        // Set button labels or other properties as needed
        saveButton.setText("Save");
        cancelButton.setText("Cancel");
        deleteButton.setText("Delete");

        // Add event listeners or click handlers for the buttons if needed
        saveButton.addClickListener(e -> handleSaveButtonClick());
        cancelButton.addClickListener(e -> handleCancelButtonClick());
        deleteButton.addClickListener(e -> handleDeleteButtonClick());
    }

    public void handleSaveButtonClick() {
        // Handle the Save button click event here
        // You can dispatch a custom event or perform other actions
    }

    public void handleCancelButtonClick() {
        // Handle the Cancel button click event here
        // You can dispatch a custom event or perform other actions
    }

    public void handleDeleteButtonClick() {
        // Handle the Delete button click event here
        // You can dispatch a custom event or perform other actions
    }
}

File: frontend/src/js/form-buttons-bar.js
import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';

class FormButtonsBar extends PolymerElement {
    static get template() {
        return html`
            <style>
                /* Define your component's styles here */
                .button-bar {
                    display: flex;
                    justify-content: space-between;
                }
            </style>
            <div class="button-bar">
                <button on-click="_handleSave" id="saveButton">Save</button>
                <button on-click="_handleCancel" id="cancelButton">Cancel</button>
                <button on-click="_handleDelete" id="cancelButton">Delete</button>
            </div>
        `;
    }

    static get is() {
        return 'form-buttons-bar';
    }

    _handleSave() {
        // Handle the Save button click event here
        this.dispatchEvent(new CustomEvent('save'));
    }

    _handleCancel() {
        // Handle the Cancel button click event here
        this.dispatchEvent(new CustomEvent('cancel'));
    }

    _handleDelete() {
        // Handle the Delete button click event here
        this.dispatchEvent(new CustomEvent('delete'));
    }
}

customElements.define(FormButtonsBar.is, FormButtonsBar);
